closure/templates/soy_plugin.bzl

```diff
@@ -0,0 +1,44 @@
+# Copyright 2019 The Closure Rules Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SoyPluginInfo = provider()
+
+def _impl(ctx):
+    files = []
+    classpath = []
+    for dep in ctx.attr.deps:
+        for jar in dep[JavaInfo].transitive_runtime_deps.to_list():
+            classpath.append(jar.path)
+            files.append(jar)
+    return [SoyPluginInfo(
+        module = ctx.attr.module,
+        deps = ctx.attr.deps,
+        classpath = depset(direct=classpath),
+        files = files,
+    )]
+
+soy_plugin = rule(
```

Comment:
I wasn't sure if `soy_plugin` was a good name; the rest of the rules say `closure_template_`...but then I wasn't sure if it should be `closure_template_plugin` or `closure_templates_plugin`.

Reply:
Right, the new rules are all called `soy_*`, so let's use the old naming convention for this. I have a small preference for `closure_templates_plugin`, but either name is fine :).

Reply:
Updated to closure_templates_plugin